### PR TITLE
WIP - [FIX #4327] Migrate to react-native-firebase

### LINKED ISSUE #4327: Migrate from `react-native-fcm` to `react-native-firebase`
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a developer, I want to upgrade from `react-native-fcm` to `react-native-firebase` so that I have access to the latest APIs and bugfixes.

The `react-native-fcm` package is no longer guaranteed to have any active development, while `react-native-firebase` has great backing:

image

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: Follow steps in https://github.com/evollu/react-native-fcm#to-existing-react-native-fcm-users to upgrade current setup to `react-native-firebase`

### PR DESCRIPTION
**Pull request by rasom**
this is just rebased https://github.com/status-im/status-react/pull/4446
i will close this PR after testing (just don't want to touch original codebase until we sure that i rebased everything correctly)

status: ready

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -211,7 +212,7 @@ dependencies {
     implementation "com.android.support:appcompat-v7:26.1.0"
     // Force using exact RN version instead of relying on gradle dependency resolution
     // https://docs.gradle.org/current/userguide/introduction_dependency_management.html#sec:dependency_resolution
-    compile ("com.facebook.react:react-native:0.55.4") { force = true }  // From node_modules
+    implementation ("com.facebook.react:react-native:0.53.3") { force = true }  // From node_modules
```
Comment (pedropombeiro) [inline android/app/build.gradle]: Looks like a merge issue, this should stay at 0.55.4 right?

**Review by pedropombeiro — CHANGES_REQUESTED** [review]
Found just a merge issue, but compared with my own rebase and all else looks the same.

File: android/app/build.gradle (hunk 1 above)
Comment (rasom) [reply android/app/build.gradle]: fixed, thanks

File: android/app/build.gradle [hunk 2]
```diff
@@ -212,7 +212,7 @@ dependencies {
     implementation "com.android.support:appcompat-v7:26.1.0"
     // Force using exact RN version instead of relying on gradle dependency resolution
     // https://docs.gradle.org/current/userguide/introduction_dependency_management.html#sec:dependency_resolution
-    implementation ("com.facebook.react:react-native:0.53.3") { force = true }  // From node_modules
+    compile ("com.facebook.react:react-native:0.55.4") { force = true }  // From node_modules
```
Comment (pedropombeiro) [inline android/app/build.gradle]: Don't we want to use `implementation` though? From original PR: https://github.com/status-im/status-react/pull/4446#discussion_r190916069

**Review by pedropombeiro — APPROVED** [review]
Looks good, now it's just a question of testing (which I imagine will be good since I tested recently with a similar rebase) and also having your opinion on the actual changes of this PR, to see if we're not introducing something unnecessary to the actual issue.

File: android/build.gradle [hunk 3]
```diff
@@ -6,8 +6,8 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
-        classpath 'com.google.gms:google-services:3.0.0'
+        classpath 'com.android.tools.build:gradle:3.1.2'
```
Comment (rasom) [inline android/build.gradle:9]: not sure that this change is necessary, but not a problem

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```
Comment (rasom) [inline android/gradle/wrapper/gradle-wrapper.properties:6]: shouldn't be required as well

**Comment by rasom** [comment]
@PombeirP looks fine for me

**Comment by pedropombeiro** [comment]
Thanks @rasom!

File: android/app/build.gradle [hunk 5]
```diff
@@ -211,7 +212,7 @@ dependencies {
     implementation "com.android.support:appcompat-v7:26.1.0"
     // Force using exact RN version instead of relying on gradle dependency resolution
     // https://docs.gradle.org/current/userguide/introduction_dependency_management.html#sec:dependency_resolution
-    compile ("com.facebook.react:react-native:0.55.4") { force = true }  // From node_modules
```
Comment (jeluard) [inline android/app/build.gradle]: Previous comment applies to the use of `compile` I believe. @dmitryn Can you comment?

File: android/app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -9,6 +9,8 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
     <uses-permission android:name="android.permission.READ_PROFILE" />
+    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
```
Comment (jeluard) [inline android/app/src/main/AndroidManifest.xml:12]: Do we want those?

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (jeluard) [reply android/gradle/wrapper/gradle-wrapper.properties:6]: Can we remove it?

File: ios/Podfile.lock [hunk 7]
```diff
@@ -1,57 +1,90 @@
...
+  React: aa2040dbb6f317b95314968021bd2888816e03d5
+  react-native-background-timer: 63dcbf37dbcf294b5c6c071afcdc661fa06a7594
+  RNKeychain: 627c6095cef215dd3d9804a9a9cf45ab96aa3997
+  yoga: a23273df0088bf7f2bb7e5d7b00044ea57a2a54a
 
-PODFILE CHECKSUM: 653ad2d673bf8aa8127325a078e07aedc8dbe463
+PODFILE CHECKSUM: 018eba4ceb8d7d516fe0130fab47bded597e9d44
 
-COCOAPODS: 1.3.1
+COCOAPODS: 1.5.3
```
Comment (jeluard) [inline ios/Podfile.lock:90]: Is this required?

File: ios/Podfile.lock [hunk 8]
```diff
@@ -1,57 +1,90 @@
...
+    - GoogleToolboxForMac/Defines (= 2.1.4)
+  - "GoogleToolboxForMac/NSData+zlib (2.1.4)":
+    - GoogleToolboxForMac/Defines (= 2.1.4)
+  - nanopb (0.3.8):
+    - nanopb/decode (= 0.3.8)
+    - nanopb/encode (= 0.3.8)
+  - nanopb/decode (0.3.8)
+  - nanopb/encode (0.3.8)
+  - Protobuf (3.5.0)
+  - React (0.55.4):
```
Comment (jeluard) [inline ios/Podfile.lock:35]: Why changes to versions here?

File: ios/Podfile.lock (hunk 7 above)
Comment (rasom) [reply ios/Podfile.lock:90]: yes it is

File: android/build.gradle (hunk 3 above)
Comment (rasom) [reply android/build.gradle:9]: https://rnfirebase.io/docs/v4.2.x/installation/android

Comment (rasom) [reply android/build.gradle:9]: it is necessary

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (rasom) [reply android/gradle/wrapper/gradle-wrapper.properties:6]: https://rnfirebase.io/docs/v4.2.x/installation/android it seems this change is necessary

File: ios/Podfile.lock (hunk 8 above)
Comment (rasom) [reply ios/Podfile.lock:35]: @jeluard this file is autogenerated, hard to answer tbh. I suppose it's related to different cocoapods version

File: ios/Podfile.lock (hunk 7 above)
Comment (rasom) [reply ios/Podfile.lock:90]: Firebase deps require higher version of cocoapods

File: android/app/src/main/AndroidManifest.xml (hunk 6 above)
Comment (rasom) [reply android/app/src/main/AndroidManifest.xml:12]: https://rnfirebase.io/docs/v4.2.x/notifications/android#Update-Android-Manifest

**Review by yenda — APPROVED** [review]

**Comment by pedropombeiro** [comment]
This PR implies an upgrade to RN 0.55 which contains a regression: https://github.com/facebook/react-native/issues/19339 https://github.com/status-im/status-react/issues/4829

We need to wait for a fix before merging this.

**Comment by rasom** [comment]
@PombeirP @annadanchenko is this task still relevant o we can just close PR?

**Comment by pedropombeiro** [comment]
It looks like we're already on RN 0.55 and `react-native-firebase`, so I'd say we can close this.

**Pull request closed without merging**